mass-merge.py

```diff
@@ -0,0 +1,210 @@
+                        sys.exit(-1)
+                    first_entry=False
+                merge_name = row[merge_col]
+                ident = row['ident']
+                
+                assert ' ' not in ident, f"identifiers cannot have spaces - but '{ident}' does."
+                assert ident not in all_idents, f"duplicate identifer: '{ident}'"
+                all_idents.add(ident)
+
+                merge_d[merge_name].append(ident)
+                count += 1
+        notify(f"loaded {count} identifiers from '{filename}'")
+
+    num_merge_names = len(merge_d)
+    notify(f"found a total of {num_merge_names} distinct values for signature merging.")
+
+    # load each db and check that we can find all idents
+    found_idents = set()
+    idx_list = []
+    for db in args.dblist:
+        notify(f"loading index '{db}'")
+        idx = sourmash.load_file_as_index(db)
+
+        manifest = idx.manifest
+        if manifest is None:
+            error(f"ERROR on filename '{db}'.")
+            error("No manifest, but a manifest is required.")
+            sys.exit(-1)
+
+        ident_picklist = SignaturePicklist('ident')
+        ident_picklist.pickset = all_idents
+        check_idx = idx.select(ksize=args.ksize,
+                                moltype=moltype,
+                                picklist=ident_picklist)
+        # do we need to do any checks for duplicate idents between dbs?
+        found_idents.update(ident_picklist.found)
+        idx_list.append(check_idx)
+
+    # make sure that we get all the things.
+    if not all_idents.issubset(found_idents):
+        remaining = all_idents - found_idents
+        error(f"ERROR: {len(remaining)} identifiers from spreadsheet not found.")
+        example_missing = "\n".join(remaining)
+        error(f"Here are some examples: {example_missing}")
+        sys.exit(-1)
+
+    if args.check:
+        notify("Everything looks copacetic. Exiting as requested by `--check`")
+        sys.exit(0)
+
+    notify("Everything looks copacetic. Proceeding to merge!")
+
+    # go through, do merge, save.
+    with sourmash_args.SaveSignaturesToLocation(args.output) as save_sigs:
+        n=0
+        n_singletons = 0
+        for m, (merge_name, idents) in enumerate(merge_d.items()):
+            if m % 100 == 0:
+                merge_percent = float(n)/len(found_idents) * 100
+                notify(f"...merging sigs for {merge_name} ({merge_percent:.1f}% of sigs merged)", end="\r")
+
+            # build a new picklist for idents to be merged
+            ident_picklist = SignaturePicklist('ident')
+            ident_picklist.pickset = set(idents)
+
+            # if singleton, just rename
+            if len(idents) == 1:
+                n_singletons+=1
+                this_idx = idx_list[0].select(picklist=ident_picklist)
+                ss = next(this_idx.signatures())
+                ss._name = merge_name
+                save_sigs.add(ss)
+                n+=1
```

Comment:
@ctb - singletons were taking forever, so I decided to just handle them separately, following your `rename` protocol. Not sure how much time it'll save, but seems worthwhile. 

Any comments here? Otherwise, I think this PR is all finished!

Reply:
seems legit!

right now the picklist and `Index` code isn't super friendly for such things. This is another use case for #1848, I think, where we would be able to use internal location to retrieve the signature directly.

Reply:
yes, that would be great!

ok resolving and merging :)